### ISSUE TITLE
Api Endpoint returns wrong cultureinfo

### ISSUE THREAD

**Issue by seanhakbb** [umbraco/Umbraco-CMS#16453]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

13.3.2

### Bug summary

Our API worked fine togheter with culture host binding and CultureInfo.CurrentCulture, with latest version this stopped working.

### Specifics

In our old v13.1.0 our API worked fine using this setup:
Culture mapping:
domain.se => sv (swedish)
domain.dk => da (danish)
when accesing api via domain.se/api/endpoint, CultureInfo.CurrentCulture returned "sv" as expected

But with latest release (13.3.2) this now returns en-us, seemingly ignoring our culture binding.

### Steps to reproduce

create site with two languages
domain.se => sv (swedish)
domain.dk => da (danish)

create an api endpoint that returns CultureInfo.CurrentCulture

### Expected result / actual result

domain.se/api/endpoint should return "sv"
domain.dk/api/endpoint should return "dk"

domain.se/api/endpoint returns "en-us"
domain.dk/api/endpoint returns "en-us"

**Comment by seanhakbb** [comment]
Small update with more strange things relating to latest 13 lts release.

We have also noticed this error in a custom controller as well now.
variationContextAccessor always returns our default language (sv), even though im visiting domain.com/en which is in bindings set to use english.

This also affects dictionary values, however in this case it always try to load dictionaries using my machine default, en-US.

Outside our custom controller all pages load fine, and when hardcoding culture values the right content is loaded (_variationContextAccessor.VariationContext = new VariationContext("en");) However we need to set CurrentThread.CurrentCulture to have dictionaries working.

Another thing we noticed, Model.GetCultureFromDomains() seems to display the correct language in our views.
